src/external_data_ifaces_impl.cpp

```diff
@@ -109,4 +111,52 @@ sdbusplus::async::task<> ExternalDataIFacesImpl::fetchBMCPosition()
     co_return;
 }
 
+sdbusplus::async::task<>
+    ExternalDataIFacesImpl::createErrorLog(const std::string& errMsg,
+                                           const Level& errSeverity,
+
+                                           const json& calloutsDetails)
+{
+    try
+    {
+        data_sync::error_log::FFDCFileSet ffdcFileSet(
+
+            calloutsDetails);
+        data_sync::error_log::FFDCFileInfoSet ffdcFileInfoSet;
+        ffdcFileSet.data_sync::error_log::FFDCFileSet::transformFFDCFiles(
+            ffdcFileInfoSet);
+
+        const auto* const loggingCreateObjPath = "/xyz/openbmc_project/logging";
+        std::string service = co_await getDBusService(
```

Comment:
We could use service name from autogenerated code as per YAML (Refer https://github.com/openbmc/phosphor-dbus-interfaces/blob/master/yaml/xyz/openbmc_project/Logging/Create.interface.yaml#L102) - we could avoid DBus call - Refer existing dbus call implementation in the data sync.